cmd/gateway/initialize.go

```diff
@@ -44,30 +44,22 @@ func initialize(cfg initializeConfig) error {
 
 	depCtx, err := cfg.collector.Collect(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to collect deployment context: %w", err)
+		cfg.logger.Error(err, "error collecting deployment context")
 	}
 
 	cfg.logger.Info("Deployment context collected", "deployment context", depCtx)
```

Comment:
This should be logged if there was no error, otherwise we get both logs, which could be confusing.

Reply:
We still get a deployment context even if we encounter an error so I think it's ok to keep.